book/tutorials/swesarr/swesarr_tut.ipynb

```diff
@@ -0,0 +1,840 @@
+    "\n",
+    "<div>\n",
+    "<img src=\"https://blogs.nasa.gov/swesarr/wp-content/uploads/sites/305/2021/06/passes.png\", width=\"500\", title=\"SWESARR passes over Grand Mesa in 2020\" />\n",
+    "</div>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "df320f07-895c-45be-9859-706516fd9cea",
+   "metadata": {},
+   "source": [
+    "## Reading SWESARR Data\n",
+    "\n",
+    "- SWESARR's SAR data is organized with a common file naming convention for finding the time, location, and type of data\n",
+    "- [Lets look at the prerelease data on its homepage](https://glihtdata.gsfc.nasa.gov/files/radar/SWESARR/prerelease/)\n",
+    "<div>\n",
+    "<img src=\"https://blogs.nasa.gov/swesarr/wp-content/uploads/sites/305/2020/09/SWESARR-Naming-Convention_recolor_v2.png\" width=\"1000\"/>\n",
+    "</div>\n",
+    "\n",
+    "***\n",
+    "<br><br>"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c5d5a065-efbf-4735-b514-b259f8ea2822",
+   "metadata": {},
+   "source": [
+    "<CENTER>\n",
+    "<H1 style=\"color:red\">\n",
+    "Accessing Data: SAR\n",
+    "</H1>\n",
+    "</CENTER>\n",
+    "\n",
+    "### SAR Data Example"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "7bf5bc1a-e9eb-4813-a4aa-5205a8762ebe",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Import several libraries. \n",
+    "# comments to the right could be useful for local installation on Windows.\n",
+    "\n",
+    "from shapely import speedups      # https://www.lfd.uci.edu/~gohlke/pythonlibs/\n",
+    "#speedups.disable()                # <-- handle a potential error in cartopy\n",
+    "\n",
+    "import requests                   # !conda install -c anaconda requests \n",
+    "\n",
+    "# raster manipulation libraries\n",
+    "import rasterio                   # https://www.lfd.uci.edu/~gohlke/pythonlibs/\n",
+    "from osgeo import gdal            # https://www.lfd.uci.edu/~gohlke/pythonlibs/\n",
+    "import cartopy.crs as ccrs        # https://www.lfd.uci.edu/~gohlke/pythonlibs/\n",
+    "import rioxarray as rxr           # !conda install -c conda-forge rioxarray\n",
+    "import xarray as xr               # !conda install -c conda-forge xarray dask netCDF4 bottleneck\n",
+    "\n",
+    "# SnowEx Pit Data!\n",
+    "from snowexsql.api import LayerMeasurements # pip install snowexsql\n",
+    "\n",
+    "# plotting tools\n",
+    "from matplotlib import pyplot     # !conda install matplotlib\n",
+    "import datashader as ds           # https://www.lfd.uci.edu/~gohlke/pythonlibs/\n",
+    "import hvplot.xarray              # !conda install hvplot\n",
+    "import hvplot.pandas\n",
+    "import holoviews as hv\n",
+    "\n",
+    "# append the subfolders of the current working directory to pythons path\n",
+    "import os, sys, glob\n",
+    "\n",
+    "swesarr_subdirs = [\"data\", \"util\"]\n",
+    "tmp = [sys.path.append(os.getcwd() + \"/\" + sd) for sd in swesarr_subdirs]\n",
```

Comment:
for tutorial data on cryocloud you'll likely have better performance downloading to `/tmp` (https://docs.2i2c.org/user/topics/data/)